libpromises/pipes_unix.c

```diff
@@ -448,7 +463,7 @@ FILE *cf_popensetuid(const char *command, const char *type, uid_t uid, gid_t gid
             }
         }
 
-        CloseChildrenFD();
+        CloseChildrenFDUnsafe();
```

Comment:
Hmmm the code after this in some error conditions returns NULL instead of doing `_exit()` or `exec()`, so calling this unsafe function could be troublesome. TODO investigate.

Reply:
This seems like a separate bug though, the forked process should never return from this function.

Reply:
Separate bug indeed, fixed in 39605fc1c3f913fb50ff24cfc9194f7f59900bcd.